src/common/AdaptiveDecrypter.h

```diff
@@ -0,0 +1,24 @@
+#include <bento4/Ap4.h>
+
+enum class ENCRYPTION_SCHEME
+{
+  NONE,
+  CENC,
+  CBCS
+};
+
+class Adaptive_CencSingleSampleDecrypter : public AP4_CencSingleSampleDecrypter
+{
+public:
+  
+  Adaptive_CencSingleSampleDecrypter() : AP4_CencSingleSampleDecrypter(0) {};
+
+  void SetCrypto(AP4_UI08 cryptBlocks, AP4_UI08 skipBlocks)
+  {
+    m_CryptBlocks = cryptBlocks;
+    m_SkipBlocks = skipBlocks;
+  };
+  virtual void SetEncryptionScheme(ENCRYPTION_SCHEME encryptionScheme){};

```

Comment:
```suggestion
  virtual void SetEncryptionScheme(ENCRYPTION_SCHEME encryptionScheme) = 0;
```
this is pure virtual

Reply:
The intention here was to have a default implementation in the case of a future module that doesn't need this info, therefore not having to override just to put `{}` in the derived class.

Reply:
In the interest of expedience though I'll change it :)

Reply:
in this case i agree keep it as is